app/src/main/java/org/simple/clinic/bp/entry/BloodPressureEntrySheet.kt

```diff
@@ -89,7 +91,12 @@ class BloodPressureEntrySheet : BottomSheetActivity() {
   private fun sheetCreates(): Observable<UiEvent> {
     val patientUuid = intent.extras!!.getSerializable(KEY_UUID) as UUID
     val openAs = intent.extras!!.getSerializable(KEY_OPEN_AS) as OpenAs
-    return Observable.just(BloodPressureEntrySheetCreated(openAs, patientUuid))
+    return Observable
+        .just(BloodPressureEntrySheetCreated(openAs, patientUuid))
+        // This delay stops the race condition (?) that happens frequently with replay().refCount()
+        // in the controller. Temporary workaround until we figure out what exactly is going on.
+        .delay(100L, TimeUnit.MILLISECONDS)
```

Comment:
Time based workarounds are not very good workarounds :p Did you not consider manually adding an `observeOn(io())` before your database call?

Reply:
Right! I'll do that instead.

Reply:
Tried this out and it doesn't seem to work. What I did notice though, was that the db query method was being invoked twice, once on the IO thread and once more on the main thread.

Leaving this workaround in for now until we figure out why the other problem actually happens.